python/ray/node.py

```diff
@@ -718,7 +721,7 @@ def _kill_process_type(self,
             if wait:
                 process.wait()
 
-        del self.all_processes[process_type]
+        self.all_processes.pop(process_type, None)
```

Comment:
What's the reason for changing this?

Reply:
It was erroring due to it being missing because we're not launching the reaper when we use this kernel mechanism.

Reply:
This seems somewhat error-prone - can we just check for a condition and not call into this code when the reaper isn't spawned/won't be in the list (same as above)?

Reply:
Done! Pushed.